mhs-reference-implementation/mhs/sender/sender.py

```diff
@@ -22,54 +21,80 @@ def __init__(self, interaction_name):
 
 
 class Sender:
-    def __init__(self, interactions_config, message_builder, transport):
+    def __init__(self, interactions_config, message_builder, transport, party_id):
         """Create a new Sender that uses the specified dependencies to load config, build a message and send it.
 
         :param interactions_config: The object that can be used to obtain interaction-specific configuration details.
         :param message_builder: The message builder that can be used to build a message.
         :param transport: The transport that can be used to send messages.
+        :param party_id: The party ID of this MHS. Sent in ebXML requests.
         """
 
         self.interactions_config = interactions_config
         self.message_builder = message_builder
         self.transport = transport
+        self.party_id = party_id
 
-    def send_message(self, interaction_name, message_to_send):
-        """Send the specified message for the interaction named.
+    def build_message(self, interaction_name: str, content: str) -> Tuple[str, str]:
+        """Build a message for the specified interaction. Wraps the provided content if required.
 
         :param interaction_name: The name of the interaction the message is related to.
-        :param message_to_send: A string representing the message to be sent.
-        :return: The content of the response received when the message was sent.
+        :param content: The message content to be sent.
+        :return: A tuple containing the ID of the message (if an asynchronous response is expected) and the content of
+        the response received. The message ID will be None if no asynchronous response is expected.
+        """
+        interaction_details = self._get_interaction_details(interaction_name)
+
+        async_pattern = interaction_details[ASYNC_RESPONSE_EXPECTED]
```

Comment:
I feel like the name of this variable doesn't give me an indication that what I am expecting here is a bool. Would something like `is_async` or `is_async_pattern` make it bit more clear?

Reply:
Fair point, I'll go with `is_async`.